gammapy/modeling/models/spatial.py

```diff
@@ -775,6 +775,38 @@ def to_region(self, **kwargs):
             **kwargs,
         )
 
+    @classmethod
+    def from_region(cls, region, **kwargs):
+        """Create a `DiskSpatialModel from a ~regions.EllipseSkyRegion`
+
+        Parameters
+        ----------
+        region : `~regions.EllipseSkyRegion` or ~regions.CircleSkyRegion`
+            region to create model from
+
+        Returns
+        -------
+        spatial_model : `~gammapy.modeling.models.DiskSpatialModel`
+        """
+        if isinstance(region, CircleSkyRegion):
+            region = EllipseSkyRegion(
+                center=region.center, width=region.radius, height=region.radius
+            )
+        if not isinstance(region, EllipseSkyRegion):
+            raise TypeError("Please provide a Circular or Elliptic region")
+        frame = kwargs.pop("frame", region.center.frame.name)
```

Comment:
if the given `frame` is different from the region frame the rotation angle seems incorrect.  Here is an example:


```
from regions import EllipseSkyRegion
from astropy.coordinates import SkyCoord
import astropy.units as u
from gammapy.modeling.models import DiskSpatialModel
import matplotlib.pyplot as plt

region = EllipseSkyRegion(
    center=SkyCoord(0, -10, unit="deg"),
    height=0.3 * u.deg,
    width=1.0 * u.deg,
    angle=0 * u.deg,
)
kwargs={}
model = DiskSpatialModel.from_region(region, **kwargs)
plt.figure()
ax = model.plot(add_cbar=True)

# illustrate size parameter
plot_region = model.to_region().to_pixel(ax.wcs)
artist = plot_region.as_artist(facecolor="none", edgecolor="b", lw=2)
ax.add_artist(artist)

kwargs={"frame":"galactic"}
model = DiskSpatialModel.from_region(region, **kwargs)

# illustrate size parameter
plot_region = model.to_region().to_pixel(ax.wcs)
artist = plot_region.as_artist(facecolor="none", edgecolor="m", lw=2)
ax.add_artist(artist)
```
![image](https://user-images.githubusercontent.com/25845654/214543792-516f459a-ba3e-481b-b9b0-2b7ee540ec57.png)

Reply:
I think you need to use a different `ax` here for the 2 cases. Plotting on the same figure is creating the apparent issue,

Otherwise,

```
from regions import EllipseSkyRegion
from astropy.coordinates import SkyCoord
import astropy.units as u
from gammapy.modeling.models import DiskSpatialModel
import matplotlib.pyplot as plt

region = EllipseSkyRegion(
    center=SkyCoord(0, -10, unit="deg"),
    height=0.3 * u.deg,
    width=1.0 * u.deg,
    angle=-40.0 * u.deg,
)
kwargs={}
model = DiskSpatialModel.from_region(region, **kwargs)
plt.figure()
ax = model.plot(add_cbar=True)
plot_region = model.to_region().to_pixel(ax.wcs)
artist = plot_region.as_artist(facecolor="none", edgecolor="b", lw=2)
ax.add_artist(artist)
```

gives 

![image](https://user-images.githubusercontent.com/32677370/214589150-f2c4dada-afc5-4e0e-b729-d329f079ea68.png)

and
```
# change frame
region = EllipseSkyRegion(
    center=SkyCoord(0, -10, unit="deg"),
    height=0.3 * u.deg,
    width=1.0 * u.deg,
    angle=0 * u.deg,
)

kwargs={"frame":"galactic"}
model = DiskSpatialModel.from_region(region, **kwargs)

ax = model.plot()
# illustrate size parameter
plot_region = model.to_region().to_pixel(ax.wcs)
artist = plot_region.as_artist(facecolor="none", edgecolor="m", lw=2)
ax.add_artist(artist)
```

gives 
![image](https://user-images.githubusercontent.com/32677370/214589261-625f7a61-e8cc-4958-a270-57f4a268130b.png)

Reply:
but if the region are equivalent `model.to_region().to_pixel(ax.wcs)` should do the projection correctly for any `ax.wcs`

Reply:
yes indeed @QRemy ! The frame transfer is incorrect in this PR. I will fix this. Thanks a lot